p32/src/bank.rs

```diff
@@ -0,0 +1,442 @@
+#[derive(Debug, PartialEq)]
+pub struct User {
+    name: String,
+    credit_line: u64,
+    balance: i64,
+}
+
+trait BalanceAccessor {
+    fn get_balance(&self) -> i64;
+    fn set_balance(&mut self, value: i64);
+}
+
+impl BalanceAccessor for User {
+    fn get_balance(&self) -> i64 {
+        self.balance
+    }
+
+    fn set_balance(&mut self, value: i64) {
+        self.balance = value
+    }
+}
+
+impl User {
+    // type of name?
+    pub fn new(name: String, credit_line: u64, balance: i64) -> Self {
+        Self {
+            name,
+            credit_line,
+            balance,
+        }
+    }
+}
+
+impl MaxCredit for User {
+    fn max_credit(&self) -> u64 {
+        self.balance as u64 + self.credit_line
+    }
+}
+
+trait Balance {
+    fn balance(&self) -> BalanceSheet;
+}
+
+trait MaxCredit {
+    fn max_credit(&self) -> u64;
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Bank {
+    users: Vec<User>,
+    name: String,
+    credit_interest: u64,
+    debit_interest: u64,
+}
+
+impl Bank {
+    pub(crate) fn merge(&mut self, another: Bank) -> bool {
+        for user_from_new_bank in another.users {
+            let user_name = &user_from_new_bank.name;
+            match self.get_user_index_by_id(user_name) {
+                None => self.users.push(user_from_new_bank),
+                Some(overlapping_user_index) => {
+                    let old_balance = self.users[overlapping_user_index].balance;
+                    let new_balance = old_balance + user_from_new_bank.balance;
+                    self.users[overlapping_user_index].set_balance(new_balance)
```

Comment:
You can write it as `self.users[overlapping_user_index] += user_from_new_bank.balance;`